plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -562,28 +550,12 @@ public void testTableComments()
                 ")";
         assertUpdate(format(createTableWithoutComment, format));
         assertUpdate("COMMENT ON TABLE test_table_comments IS NULL");
-        MaterializedResult resultOfRemovingComment = computeActual(showCreateTable);
+        MaterializedResult resultOfRemovingComment = computeActual("SHOW CREATE TABLE test_table_comments");
         assertEquals(getOnlyElement(resultOfRemovingComment.getOnlyColumnAsSet()), format(createTableWithoutComment, format));
 
         dropTable("iceberg.tpch.test_table_comments");
     }
 
-    // TODO: This test shows that column $snapshot_id doesn't exist at this time.  Decide if we should
-    // add it and $snapshot_timestamp_ms
-    @Test(enabled = false)
-    public void testQueryBySnapshotId()
-    {
-        assertUpdate("CREATE TABLE test_query_by_snapshot (col0 INTEGER, col1 BIGINT)");
-        assertUpdate("INSERT INTO test_query_by_snapshot (col0, col1) VALUES (123, CAST(987 AS BIGINT))", 1);
-        long afterFirstInsertId = getLatestSnapshotId("test_query_by_snapshot");
-
-        assertUpdate("INSERT INTO test_query_by_snapshot (col0, col1) VALUES (456, CAST(654 AS BIGINT))", 1);
-        assertQuery("SELECT * FROM test_query_by_snapshot ORDER BY col0", "VALUES (123, CAST(987 AS BIGINT)), (456, CAST(654 AS BIGINT))");
-        assertQuery("SELECT * FROM test_query_by_snapshot WHERE \"$snapshot_id\" = " + afterFirstInsertId, "VALUES (123, CAST(987 AS BIGINT))");
```

Comment:
not sure what the original test was intended to do, but we can replace this with 
```
        assertQuery("SELECT * FROM \"test_query_by_snapshot@" + afterFirstInsertId + "\"", "VALUES (123, CAST(987 AS BIGINT))");
```

I'm on the fence, as there are no current tests to demonstrate that this feature works as expected. for example, when multiple snapshots for the same table are accessed in the same query. (And the test seems to have added with a doc, and not with feature)

Reply:
@phd3 good catch!

let's make sure we have good test coverage for that.
Can you please file an issue?

This will help parallelize work.

Reply:
https://github.com/trinodb/trino/issues/8663